homeassistant/helpers/event.py

```diff
@@ -1362,14 +1363,20 @@ def async_call_later(
     delay: float | timedelta,
     action: HassJob[[datetime], Coroutine[Any, Any, None] | None]
     | Callable[[datetime], Coroutine[Any, Any, None] | None],
+    *,
+    cancel_on_hass_stop: bool = False,
 ) -> CALLBACK_TYPE:
     """Add a listener that is called in <delay>."""
     if isinstance(delay, timedelta):
         delay = delay.total_seconds()
 
+    _unsub_cancel_on_hass_stop: CALLBACK_TYPE | None = None
+
     @callback
     def run_action(job: HassJob[[datetime], Coroutine[Any, Any, None] | None]) -> None:
         """Call the action."""
+        if _unsub_cancel_on_hass_stop:
+            _unsub_cancel_on_hass_stop()
```

Comment:
It should be unsubbed after the task runs? Otherwise we only cancel the time registration. Not the task itself.

Reply:
Is there any reason not to cancel the actual task? Im okey with this way too just pondering.

Reply:
Ah, we dont have access to any task with call later. So nevermind.

Reply:
I'm not sure I follow. Do you have an implementation in mind?

Reply:
🤣